### PR TITLE
Ignore case

### PR DESCRIPTION
**Pull request by madmuffin1**
I created two new rules in order to compare fields in a case insensitive manner:

- `$startsWithIgnoreCase`: compares the prefix
- `$ignoreCase`: check for case insensitive identity

### CODE REVIEW

File: startswithInsensitive.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	}, nil
+}
+
+func (swt *startsWithInsensitiveTest) Run(input interface{}) error {
+	castedString, ok := castString(input)
+	if !ok {
+		return fmt.Errorf("input (%#v %v) isn't a string", input, reflect.TypeOf(input))
+	}
+
+	if !strings.HasPrefix(strings.ToLower(castedString), strings.ToLower(swt.expectedStart)) {
```
Comment (LennyPenny) [inline startswithInsensitive.go]: `strings.ToLower` doesn't do unicode case-folding which is used in the other test added in this PR
I think we should be consistent here

Comment (LennyPenny) [inline startswithInsensitive.go]: also I think we can move `strings.ToLower(swt.expectedStart)` to line 20

Comment (madmuffin1) [reply startswithInsensitive.go]: I tested this and it even supports title cased strings (see https://www.compart.com/en/unicode/category/Lt) so the simple ToLower/ToUpper should be fine

**Comment by madmuffin1** [comment]
@LennyPenny any more comments on this?

File: startswithInsensitive.go (hunk 1 above)
Comment (LennyPenny) [reply startswithInsensitive.go]: if we use `ToLower` here, then `EqualFold` shouldn't be used in the other test, as the implementation might lead to different results in edge cases
